services/publicapi/test/public_api_test.go

```diff
@@ -110,13 +112,54 @@ func newPublicApiConfig(txTimeout time.Duration) publicapi.Config {
 	return cfg
 }
 
+func TestSendTransaction_TimesOut(t *testing.T) {
+	test.WithContext(func(ctx context.Context) {
+		logger := log.GetLogger().WithOutput(log.NewOutput(os.Stdout).WithFormatter(log.NewHumanReadableFormatter()))
+		txpMock := makeTxMock()
+		vmMock := &services.MockVirtualMachine{}
+
+		timeoutDuration := 1 * time.Millisecond
+
+		cfg := newPublicApiConfig(timeoutDuration)
+		papi := publicapi.NewPublicApi(ctx, cfg, txpMock, vmMock, logger)
+
+		blockTime := primitives.TimestampNano(time.Now().Nanosecond())
+		txb := builders.Transaction().Builder()
+		txpMock.When("AddNewTransaction", mock.Any).Times(1).
```

Comment:
take a look at master for this code, @noambergIL introduced a test harness, make sure you adapt to it

Reply:
Done.